tests/integration/test_api.py

```diff
@@ -470,3 +477,45 @@ def test_annotate_insight_anonymous_then_authenticated(client, mocker):
         server_domain=settings.OFF_SERVER_DOMAIN,
         auth=OFFAuthentication(username="a", password="b"),
     )
+
+
+def test_get_images(client):
+    result = client.simulate_get("/api/v1/images?page=1&count=25")
+    assert result.status_code == 200
+
+    image_model_factory1 = ImageModelFactory(barcode="123")
+    image_model_factory2 = ImageModelFactory(barcode="456")
+
+    image_prediction_factory1 = ImagePredictionFactory(image=image_model_factory1.id)
+    image_prediction_factory2 = ImagePredictionFactory(image=image_model_factory2.id)
+    image_prediction_factory3 = ImagePredictionFactory()
+
+    # test with "barcode" filter
+    image_model_data = get_images(barcode="123")
+    image_model_items = [item.to_dict() for item in image_model_data]
+    assert len(image_model_items) == 1
+    assert image_model_items[0]["id"] == image_model_factory1.id
+    assert image_model_items[0]["barcode"] == "123"
```

Comment:
This test will work only if you create image_model3 without predictions as mentioned above.

And in thi case it will be image3.id that you get.